hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/source/IncrementalInputSplits.java

```diff
@@ -213,7 +213,7 @@ public Result inputSplits(
     }
 
     HoodieTableFileSystemView fsView = new HoodieTableFileSystemView(metaClient, commitTimeline, fileStatuses);
-    final String endInstant = instantToIssue.getTimestamp();
+    final String endInstant = instantRange == null ? fsView.getLastInstant().get().getTimestamp() : instantToIssue.getTimestamp();
     final AtomicInteger cnt = new AtomicInteger(0);
```

Comment:
I'm confused, when `instantRange == null` is true, that means the user first time query from earliest, and we should   read to the latest commit instant, which should be `instantToIssue` here.

Reply:
Good point from Danny.
@cuibo01 Can you please explain that? Is it still a vlid scenario?